### ISSUE TITLE
Markdown Export

### ISSUE THREAD

**Issue by mhils** [mitmproxy/pdoc#308]
hey any chance we can create a ticket for the markdown output support? I saw there is one that was closed but the markdown branch is over a hundred commits behind and I would love to track it properly :)

there's quite a bit wrong with simply renaming html files to md files right now. I am using mkdocs but, for example, the search feature of mkdocs will pick up on the html and show a lot of nonsense in the results because of it. I'm also struggling with links not working because of differences in how mkdocs treats files as directories in the URL but pdoc links to html files (that do not exist). I think proper markdown output support would make this challenge at least a tiny bit easier since the jinja templates should be simpler to modify to fit any weird mkdocs→pdoc linking issues.

thanks for the work!

_Originally posted by @Zeelot in https://github.com/mitmproxy/pdoc/issues/203#issuecomment-949286955_

**Comment by mhils** [comment]
@Zeelot made some great progress in https://github.com/codeghetti/seagulls-py/pull/39 - I think we have a reasonable template to start with. 🥳 

I have a PhD thesis to write at the moment, so it will take me a while to do the rest of the pdoc lifting. If someone needs this urgently, @Zeelot's project has the relevant template files which you can just drop in for the HTML one to get Markdown output. :)

**Comment by mhils** [comment]
I have pondered on this for a quite a bit over the last few months and have come to the conclusion that I'd like to declare pure Markdown export as out of scope for the time being. Here's my reasoning:

 - The HTML export is in many ways superior to what raw Markdown can potentially achieve. For example, we now have syntax highlighting in the function definitions, which is hard to achieve with Markdown. Having only one output format allows us to specialize on that and exploit its benefits.
 - The vast majority of pdoc user that I'm aware of uses pdoc as a standalone documentation system with some adjustments to the default template. There is only a small subset of users that embeds pdoc's docs into other content systems (mkdocs, hugo, ...). For the small subset which embeds pdoc, the majority is better served by [embedding the generated HTML](https://pdoc.dev/docs/pdoc.html#integrate-pdoc-into-other-systems) instead of Markdown. The remaining users are mostly on mkdocs, which does not take raw HTML very well. For this use case, @pawamoy et al.'s [mkdocstrings](https://mkdocstrings.github.io/) is a wonderful alternative that integrates well into the ecosystem. Long story short, raw Markdown output is really a niche use case.
 - A second output format requires a significant amount of maintenance, which is at odds with keeping this project sustainable.

Long story short, as a maintainer I have the unfortunate duty of making some hard calls, and not adding Markdown output is one of them. @Zeelot's template will continue to function for those who want this functionality, but it's not becoming part of pdoc itself.

**Comment by patrick91** [comment]
@mhils thanks for the update!
